inc/sitemaps/class-sitemap-cache-data.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * @package WPSEO\XML_Sitemaps
+ */
+
+/**
+ * Sitemap Cache Data object, manages sitemap data stored in cache
+ */
+class WPSEO_Sitemap_Cache_Data implements WPSEO_Sitemap_Cache_Data_Interface, Serializable {
+
+	/**
+	 * @var string Sitemap XML data.
+	 */
+	private $sitemap = '';
+	/**
+	 * @var string Status of the sitemap, usable or not.
+	 */
+	private $status = self::UNKNOWN;
+
+	/**
+	 * Set the sitemap XML data
+	 *
+	 * @param string $sitemap XML Content of the sitemap.
+	 */
+	public function set_sitemap( $sitemap ) {
+
+		if ( ! is_string( $sitemap ) ) {
+			$sitemap = '';
+		}
+
+		$this->sitemap = $sitemap;
+
+		/**
+		 * Empty sitemap is not usable.
+		 */
+		$this->is_usable( ! empty( $sitemap ) );
+	}
+
+	/**
+	 * Set the status of the sitemap, is it usable.
+	 *
+	 * @param bool|string $valid Is the sitemap valid or not.
+	 *
+	 * @return string The status that was set.
+	 */
+	public function set_status( $valid ) {
```

Comment:
The return value is to be able to quickly check if the setting went as expected.